lib/spack/spack/compilers/arm.py

```diff
@@ -35,7 +36,21 @@ class Arm(spack.compiler.Compiler):
     # InstalledDir:
     # /opt/arm/arm-hpc-compiler-19.0_Generic-AArch64_RHEL-7_aarch64-linux/bin
     version_argument = '--version'
-    version_regex = r'Arm C\/C\+\+\/Fortran Compiler version ([^ )]+)'
+    version_regex = r'Arm C\/C\+\+\/Fortran Compiler version ([\d\.]+) '\
+                    r'\(build number (\d+)\) '
+
+    @classmethod
+    def extract_version_from_output(cls, output):
+        """Extracts the version from compiler's output."""
+        match = re.search(cls.version_regex, output)
+        if match:
+            if match.group(1).count('.') == 1:
+                temp = match.group(1) + ".0." + match.group(2)
+            else:
```

Comment:
Can you add a second test parameterization where this else clause gets hit?

Reply:
The Arm compiler is generally released twice per year. June for ISC and November for SC. These releases are either a major (e.g. 20.0) or minor (20.1) release. The rare occasion does arise that there is a release between these events and a minor minor version is justified (e.g. 20.1.1).

When discussing the build feature detection with the compiler team, they opted for this approach in case a minor minor version is needed. None of the current releases have a minor minor designation, but this provides flexibility for the future.

Reply:
In that case, maybe you can add a fake test so we know it works and get test coverage.

Reply:
I added a fake test so if the version output is "...19.3.2 (build number 75)..." the test result is "19.3.2.75" consistent with the new version function. Both the 'if' and 'else' now have a sample test.